sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -146,6 +150,7 @@ def test_partial_dependecy_input():
     assert_raises(ValueError, partial_dependence, clf, [0], grid=grid)
 
 
+@ignore_warnings(category=DeprecationWarning)
 @pytest.mark.filterwarnings('ignore: Using or importing the ABCs from')
 # matplotlib Python3.7 warning
 @if_matplotlib
```

Comment:
I think that this is the perfect time to remove this decorator which was introduced before pytest (7 years ago). It is only used with the partial dependence.

We can introduce the following:

```python
def has_matplotlib():
    try:
        # feel free to keep the previous content of the function
        import matplotlib
        return False
    except ImportError:
        return True
...
    skip_if_no_matplotlib = pytest.mark.skipif(
        has_matplotlib(), reason="matplotlib is not installed"
    )
```

We can then use

```python
@skip_if_no_matplotlib
def test_function_to_be_skipped():
    ...
```

An additional reason (which is the main one, IMO) is that we can use it within parametrize and we can factorize code. See the example below.

Reply:
I don't think it makes sense to change the tests with this new decorator here. They will all be removed in .23.